### ISSUE TITLE
WSO2 Call mediator behaving differently in blocking vs non-blocking mode

### ISSUE THREAD

**Issue by sreenath-centime** [wso2/product-ei#5171]
WSO2 Call mediator when used to make a http endpoint request, the behaviour is different in blocking mode and non-blocking mode.

I am using wso2 Integration Studio 7.0 version to build my mediation flows.

I see this inconsistency while using call mediator in blocking mode and non-blocking mode.
In blocking mode, I am not able to catch the individual http codes like 401,403,501,503 etc and when the call fails, the control directly goes to the fault sequence defined.
To overcome this problem, I switched to non-blocking mode, but my mediation flow completes and I am able to catch the exception, but surprisingly, I see one particular error in the logs,
[2020-06-30 01:22:41,966] ERROR {org.apache.synapse.mediators.builtin.PropertyMediator} - Unknown type : INTEGER for the property mediator or the property value cannot be converted into the specified type. java.lang.NumberFormatException: For input string: ""

Can you please help me resolve this issue, either in blocking mode or non-blocking mode. 
Basically I am trying to catch individual http error codes for better error handling in my flows.

I came across a similar issue reported in the past, but the conclusion was not clear from the comments.
https://github.com/wso2/wso2-axis2/pull/166

Thanks
Sreenath

**Comment by Vathsan** [comment]
Hi @sreenath-centime ,

You may refer the blog [1] get a better understanding of configuring HTTP status codes. You may also follow that blog to configure non-error HTTP status code in non-blocking mode. To further analyze the issue regarding the NumberFormatException could you please share the synapse configuration that you are using?

[1] https://medium.com/@malakaganga.ac/allow-http-status-code-which-is-not-in-successful-httpstatuscodefamily-when-call-mediator-with-f03af4a8c709

**Comment by sreenath-centime** [comment]
Thanks @Vathsan

**Comment by yasirjanjua** [comment]
Is there any fix planned for this?

**Comment by Vathsan** [comment]
Hi @yasirjanjua,

From the fix https://github.com/wso2/wso2-axis2/pull/201, we have introduced a property to configure non-error HTTP status code/comma separated codes for call mediator in blocking mode. 
Example:
`<property name="non.error.http.status.codes" scope="axis2" value="403" type="STRING"/>`

The above fix is available in the MI-1.2.0 release and available as a WUM update for EI-6.6.0.
